src/lib/nextWordGuesserStrategy.ts

```diff
@@ -16,57 +16,85 @@
- * 4. guess word by highest count by most frequent letters (calculate frequency list of each position, eliminate letters highest on that, total # = metric)
-
- * then pick from among these which removes the most amount of words total from the candidate list.
+ * @example For calculating the score of a guess based on its ASCII character values
+ * ```typescript
+ *  class AsciiValueGuesserStrategy extends NextWordGuesserStrategyBase {
+ *      constructor(...ctorParams: ConstructorParameters<typeof NextWordGuesserStrategyBase>) {
+ *          super(...ctorParams);
+ *      }
+ *
+ *      scoreForGuess(guess: string): number {
+ *          return sumBy(guess, (chr) => chr.charCodeAt(0));
+ *      }
+ *  }
+ * ```
+ * ...and then using this to suggest guesses:
+ * ```typescript
+ *  const wordList = new wordList(methodThatReturnsAListOfWords())
+ *  const asciiValueGuesser = new AsciiValueGuesserStrategy(wordList);
+ *  const asciiValueGuesserAfterGuessingHELLO = asciiValueGuesser.withPreviousGuess({
+ *      word: 'HELLO',
+ *      result: someFunctionThatYieldsExclusionAndMandateRulesFromGuess('HELLO')
+ *  });
+ *  expect(asciiValueGuesserAfterGuessingHELLO.getAlreadyGuessedLetters()).toEqual(['H', 'E', 'L', 'O']);
+ *  console.log('Next optimal word guess after HELLO is: ',
+ *      asciiValueGuesserAfterGuessingHELLO.guessNextWordAndScore());
+ * ```
  */
-
-export interface INextWordGuesserStrategyBase {
-    getAlreadyGuessedLetters(): string[];
-
-    guessNextWordAndScore(): WordGuessAndScore;
-
-    withPreviousGuess(candidate: WordGuessAndResult): this;
-
-    get previousGuesses(): WordGuessAndResult[];
-    get wordList(): WordList;
-}
-
-export interface IStrategyScoreMethod {
-    scoreForGuess(guess: string): number;
-}
-
-export abstract class NextWordGuesserStrategyBase implements INextWordGuesserStrategyBase {
+export default abstract class NextWordGuesserStrategyBase {
+    /**
+     * Track the previously-guessed words and their resulting exclusion/mandate rules.
+     */
     protected myPreviousGuesses: WordGuessAndResult[] = [];
 
+    /**
+     *
+     * @param myWordList - the `WordList` object to use for choosing the next optimal guess.
+     */
     constructor(protected myWordList: WordList) {}
 
+    /**
+     * Implement this with a metric for scoring for the given guess, determined by the guessing strategy in use.
+     *
+     * @param guess - The guess candidate word.
+     *
+     * @returns A score calculated by a given guessing strategy, roughly equal to the number of letter & position
+     *  combinations' worth of information that would be given by this guess in that strategy.
+     */
     abstract scoreForGuess(guess: string): number;
 
-    withPreviousGuess(candidate: WordGuessAndResult) {
+    /**
+     * Modifies the calling object to procss the given candidate guess and store its resulting rules, such as certain
+     *  letters being mandated at or excluded from certain positions.
```

Comment:
Nit: Extraneous spacing.
```suggestion
     * letters being mandated at or excluded from certain positions.
```